### PR TITLE
added a console warning when loading a config.local file

### PR DESCRIPTION
**Pull request by brentcam250**
Added a console warning when loading a config.local file.

### CODE REVIEW

**Review by beraki — DISMISSED** [review]

**Review by eduardoyuidy — APPROVED** [review]

File: src/index.ts [hunk 1]
```diff
@@ -61,6 +61,12 @@ const loadFile = (filePath: string): Record<string, unknown> => {
       // eslint-disable-next-line @typescript-eslint/no-var-requires
       const config = require(filePath);
 
+      if (filePath.includes('config.local')) {
```
Comment (iansu) [inline src/index.ts:64]: It's possible to have local config files with other names like `config.development.local.js` so we should handle those cases as well

**Review by iansu — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 1 above)
Comment (brentcam250) [reply src/index.ts]: Updated to use regex to match files. /config.+local/

**Review by beraki — APPROVED** [review]

**Review by Lewis65 — APPROVED** [review]

**Review by eduardoyuidy — DISMISSED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -61,6 +61,12 @@ const loadFile = (filePath: string): Record<string, unknown> => {
       // eslint-disable-next-line @typescript-eslint/no-var-requires
       const config = require(filePath);
 
+      if (filePath.match(/config.+local/)) {
+        console.log(
+          'WARNING: Found a local config file, be cautious that this file could be published on the build phase.'
```
Comment (iansu) [inline src/index.ts:66]: This error message seems too specific to... something. I don't really know what it's referring to but we should probably just leave the message as `WARNING: Found a local config file`. We could probably also include the name of the config file it found so something like `WARNING: Found a local config file config.development.local.js`.

**Review by iansu — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 2 above)
Comment (brentcam250) [reply src/index.ts]: Changed the error to be `WARNING: Found a local config file ${fileName} `

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 1.7.2 (March 16, 2022)
+
+- Added a warning when using config.local
```
Comment (iansu) [inline CHANGELOG.md:5]:
```suggestion
## 1.7.2 (March 23, 2022)

- Add a warning when a local config file is found
```

**Review by iansu — CHANGES_REQUESTED** [review]

**Review by eduardoyuidy — APPROVED** [review]

**Review by Lewis65 — APPROVED** [review]

**Review by iansu — APPROVED** [review]

**Review by beraki — APPROVED** [review]

**Pull request merged**